1-raw-data/mailinglists/googlegroups/googlegroups_data.go

```diff
@@ -0,0 +1,429 @@
+		//if errors.Is(err, syscall.EPIPE) {
+		//	log.Printf("HTTP string get broken pipe ignored for url: %s/n", url)
+		//} else {
+		err = fmt.Errorf("Reading http response failed: %v", err)
+		return
+	}
+
+	responseString = string(bodyBytes)
+	return
+}
+
+// Create HTTP response body and return as a dom object
+func httpDomResponse(url string) (dom *goquery.Document, err error) {
+	var response *http.Response
+
+	if response, err = http.Get(url); err != nil {
+		err = fmt.Errorf("HTTP dom response returned an error: %v", err)
+		return
+	}
+	defer response.Body.Close()
+
+	if dom, err = goquery.NewDocumentFromReader(response.Body); err != nil {
+		err = fmt.Errorf("Goquery dom conversion returned an error: %v", err)
+		return
+	}
+	return
+}
+
+// Create month year filename for topic list map
+func getFileName(matchDate string) (fileName string, err error) {
+	var tempDate time.Time
+	dateSplit := strings.Split(matchDate, "/")
+	numDigMonthDay := fmt.Sprintf("%d%d", len(dateSplit[0]), len(dateSplit[1]))
+
+	// Convert string based on 1 or 2 digit month or day
+	switch numDigMonthDay {
+	case "11":
+		if tempDate, err = time.Parse("1/2/06", matchDate); err != nil {
+			err = fmt.Errorf("End date string conversion to DateTime threw an error: %v", err)
+		}
+	case "12":
+		if tempDate, err = time.Parse("1/02/06", matchDate); err != nil {
+			err = fmt.Errorf("End date string conversion to DateTime threw an error: %v", err)
+		}
+	case "21":
+		if tempDate, err = time.Parse("01/2/06", matchDate); err != nil {
+			err = fmt.Errorf("End date string conversion to DateTime threw an error: %v", err)
+		}
+	case "22":
+		if tempDate, err = time.Parse("01/02/06", matchDate); err != nil {
+			err = fmt.Errorf("End date string conversion to DateTime threw an error: %v", err)
+		}
+	}
+	// Found error in time.Parse of 2 date year that applies 20 to anything below 69. At time of thos code, anything after 2020 is future
+	if tempDate.Year() > time.Now().Year() {
+		tempDate = tempDate.AddDate(-100, 0, 0)
+	}
+
+	fileName = fmt.Sprintf("%04d-%02d.txt", tempDate.Year(), int(tempDate.Month()))
+	return
+}
+
+// Get the total topics posted at the top of the list to track all are pulled
+func getTotalTopics(dom *goquery.Document) (totalTopics int, err error) {
+	regTotal, _ := regexp.Compile("[^<]*?([0-9]+) *- *([0-9]+) of ([0-9]+)[^<]*?")
+
+	text := dom.Find("i").Text()
+
+	if regTotal.MatchString(text) {
+		match := regTotal.FindStringSubmatch(text)
+		num1, _ := strconv.Atoi(match[1])
+		num2, _ := strconv.Atoi(match[2])
+		num3, _ := strconv.Atoi(match[3])
+		numList := []int{num1, num2, num3}
+		sort.Ints(numList)
+		totalTopics = numList[2]
+	}
+	return
+}
+
+// Get message ids from topic pages and create list of raw msg urls by month
+func getMsgIDsFromDom(org, topicId, groupName string, dom *goquery.Document) (rawMsgUrl string, err error) {
+	regTopicURL, _ := regexp.Compile(fmt.Sprintf("/d/msg/%s", groupName))
```

Comment:
If you're ignoring the error here, consider using the MustCompile variant of this call (and look for it elsewhere in the stdlib).

The idea behind these functions is that they in-line their error checking and panic if there is an error.